src/edu/wright/cs/raiderplanner/controller/AssignmentController.java

```diff
@@ -0,0 +1,200 @@
+package edu.wright.cs.raiderplanner.controller;
+
+import edu.wright.cs.raiderplanner.model.QuantityType;
+import edu.wright.cs.raiderplanner.model.Task;
+import edu.wright.cs.raiderplanner.model.Assignment;
+import edu.wright.cs.raiderplanner.model.Coursework;
+import edu.wright.cs.raiderplanner.model.Exam;
+import edu.wright.cs.raiderplanner.view.UIManager;
+import javafx.application.Platform;
+import javafx.beans.binding.BooleanBinding;
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.DatePicker;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.control.MenuItem;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableRow;
+import javafx.scene.control.TableView;
+import javafx.scene.control.TextArea;
+import javafx.scene.control.TextField;
+import javafx.scene.control.cell.PropertyValueFactory;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.GridPane;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+import java.net.URL;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.ResourceBundle;
+
+/**
+ * @author Katie
+ *
+ */
+public class AssignmentController implements Initializable {
+	private Assignment assignment;
+	private boolean success = false;
+
+	/**
+	 * Default constructor.
+	 */
+	public AssignmentController() {
+	}
+
+	/**
+	 * Constructor for an AssignmentController with an existing Assignment.
+	 *
+	 * @param assignment the Assignment which will be managed by the new controller
+	 */
+	public AssignmentController(Assignment assignment) {
+		this.assignment = assignment;
+	}
+
+	/**
+	 * @return the Assignment object being managed by this controller.
+	 */
+	public Assignment getAssignment() {
+		return this.assignment;
+	}
+
+	/**
+	 * @return true if the last submit operation succeeded, false otherwise.
+	 */
+	public boolean isSuccess() {
+		return success;
+	}
+
+	// Panes:
+	@FXML private GridPane pane;
+
+	// Buttons:
+	@FXML private Button submit;
+
+	// Text:
+	@FXML private TextField name;
+	@FXML private ComboBox<String> assignmentType;
+	@FXML private TextField weighting;
+	@FXML private DatePicker date;
+
+	// Labels:
+	@FXML private Label title;
+
+
+	/**
+	 * Handle changes to the input fields.
+	 */
+	public void handleChange() {
+		if (!this.name.getText().trim().isEmpty()
```

Comment:
Does this logic work if the fields are changed to be empty? That is, does making them empty result in a NullPointerException?